doop/ui/src/components/AppContent.jsx

```diff
@@ -16,10 +16,39 @@ import { fetchData } from "../lib/apiClient"
 import { parseError } from "../lib/helpers"
 import Highlighter from "./Highlighter"
 import Violations from "./violations/violations"
+import { fetchProxy } from "utils"
+import { useGlobalsMock, useGlobalsEndpoint } from "./StoreProvider"
 
 const AppContent = ({ id, showDebugSeverities }) => {
   const { setData, setShowDebugSeverities } = useDataActions()
   const { addMessage } = useActions()
+  const isMock = useGlobalsMock()
+  const endpoint = useGlobalsEndpoint()
+
+  useEffect(() => {
+    if (isMock) {
+      fetchProxy(`${endpoint}/templates`, {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+          Accept: "application/json",
+        },
+        ...{ mock: true },
+      })
+        .then((response) => {
+          if (!response.ok) {
+            addMessage({
+              variant: "error",
+              text: parseError(e.message),
+            })
+          }
+          return response.json()
```

Comment:
It means that an attempt will be made to convert the response to JSON in any case, even if response.ok is false?

Reply:
I tested this, return response.json() is needed in order to catch the result in the next then, as this promise is pending here and not resolved yet.